submissions/Levi123/tiny-JS/index.js

```diff
@@ -0,0 +1,98 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Levi123/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const dog = {
+   species: 'dog',
+   name: 'Rich',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'Hav-hav!',
+   friends: ['Bonya'],
+}
+
+const cat = {
+   species: 'cat',
+   name: 'Bonya',
+   gender: 'female',
+   legs: 4,
+   hands: 0,
+   saying: 'Mr-mr-mr',
+   friends: ['Rich', 'Catwoman'],
+}
+
+const female = {
+   species: 'human',
+   name: 'Olya',
+   gender: 'female',
+   legs: 4,
+   hands: 4,
+   saying: "Glory to Ukraine",
+   friends: ['Yura', 'Rich'],
+}
+
+const male = {
+   species: 'human',
+   name: 'Yura',
+   gender: 'male',
+   legs: 4,
+   hands: 4,
+   saying: "Glory to Ukraine",
+   friends: ['Olya', 'Bonya'],
+}
+
+const catWoman = {
+   species: 'human',
+   name: 'Catwoman',
+   gender: 'female',
+   legs: 4,
+   hands: 4,
+   saying: cat.saying,
+   friends: ['Bonya']
+}
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+const prop = ['species', 'name', 'gender', 'legs', 'hands', 'saying', 'friends']
+
+function printObjectNew(personToPrint){
+   const propToPrint = [];
+   prop.forEach((propOfObj) => {
+      propToPrint.push(personToPrint[propOfObj]);
+   })
+   let newStr = propToPrint.join('; ');
+   print(newStr);
```

Comment:
Think of converting property names into object's property values, then joining with `'; '`.
Make the method return the result. Currently it does two jobs: (1) convert data, (b) produce output, which  doesn't really follow Separatoin of Concerns principles.

Use `Array.map` method to convert data.

Reply:
@OleksiyRudenko did I understand correctly, I need to split this function into several?

Reply:
@Levi123 while looking through your updated code version I started wondering why is it so.
I am sorry, I apparently missed one very important word here. The comment above  should read
> You should **not** use `print` inside this method.
My apologies.
We will fix this.

Reply:
> @Levi123переглядаючи вашу оновлену версію коду, я почав дивуватися, чому це так.
> 
> Вибачте, я, мабуть, пропустив тут одне дуже важливе слово. Коментар вище має прочитати
> 
> > Ви не повинні **використовувати** всередині `print`цей метод.
> 
> Мої вибачення. Ми це виправимо.

i don't understand you